sympy/solvers/solveset.py

```diff
@@ -2918,7 +2918,7 @@ def nonlinsolve(system, *symbols):
 
     >>> from sympy import pprint
     >>> from sympy.polys.polytools import is_zero_dimensional
-    >>> a, b, c, d = symbols('a, b, c, d', real=True)
+    >>> a, b, c, d = symbols('a, b, c, d', extended_real=True)
```

Comment:
This doctest is also in the main tests and is
```python
from sympy import pprint, symbols, nonlinsolve
from sympy.polys.polytools import is_zero_dimensional
a, b, c, d = symbols('a, b, c, d', real=True)
eq1 =  a + b + c + d
eq2 = a*b + b*c + c*d + d*a
eq3 = a*b*c + b*c*d + c*d*a + d*a*b
eq4 = a*b*c*d - 1
system = [eq1, eq2, eq3, eq4]
is_zero_dimensional(system)
pprint(nonlinsolve(system, [a, b, c, d]), use_unicode=False)
```
On master this gives:
```
  -1       1               1      -1              
{(---, -d, -, {d} \ {0}), (-, -d, ---, {d} \ {0})}
   d       d               d       d           
```
However under this PR it is
```
  -1       1       1      -1      
{(---, -d, -, d), (-, -d, ---, d)}
   d       d       d       d      
```
I've changed it to extended_real to preserve the original output. I'm not sure which is actually better though.

Is the test intentionally demonstrating something with the exclusion of zero? Should I instead change the test output?

Reply:
Something is wrong, I think, because being extended doesn't change the fact that `d` cannot be 0.

Reply:
So eq4 cannot be satisfied if d=0 unless a, b or c is infinite. That would seem pertinent but the difference in behaviour is the wrong way round. When a, b, and c are declared as real (i.e. finite under this PR) it then becomes possible for d to be zero...

Reply:
The same difference occurs on master if the symbols are declared with finite=True. Perhaps a separate issue should be opened for this.

Reply:
I've opened #16909